.buildkite/pipeline.yml

```diff
@@ -1,3 +1,57 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/buildkite/pipeline-schema/main/schema.json
+
+env:
+  SETUP_MAGE_VERSION: '1.14.0'
+  SETUP_GOLANG_VERSION: '1.20.4'
+
 steps:
-  - label: "Example test"
-    command: echo "Hello!"
+  - label: ":golangci-lint: Checks formatting / linting"
+    command:
+      - ".buildkite/scripts/pre-install-command.sh 'mage -debug check'"
+    agents:
+      image: "golang:${SETUP_GOLANG_VERSION}"
+      cpu: "8"
+      memory: "4G"
+
+  - label: ":hammer: Build"
+    command:
+      - ".buildkite/scripts/pre-install-command.sh 'mage -debug build'"
+    agents:
+      image: "golang:${SETUP_GOLANG_VERSION}"
+      cpu: "8"
+      memory: "4G"
+
+  - label: "Test on Linux"
+    key: test-linux
+    command:
+      - ".buildkite/scripts/pre-install-command.sh 'mage -debug test'|tee tests-report-linux.txt"
+      - go install github.com/jstemmer/go-junit-report/v2@latest && go-junit-report > tests-report-linux.xml < tests-report-linux.txt
+      - ls
```

Comment:
Same for this, I would create a new script. So it is easy to manage PATH variable within it.